src/xo-app/new-vm/index.js

```diff
@@ -595,17 +595,16 @@ export default class NewVm extends BaseComponent {
   )
 
   _getCoresPerSocketPossibilities = createSelector(
-    () => this.props.pool,
+    () => this.props.pool.cpus.cores,
     () => this.state.state.CPUs,
-    (pool, vCPUs) => {
+    (maxCoresPerSocket, vCPUs) => {
       // According to : https://www.citrix.com/blogs/2014/03/11/citrix-xenserver-setting-more-than-one-vcpu-per-vm-to-improve-application-performance-and-server-consolidation-e-g-for-cad3-d-graphical-applications/
       const maxVCPUs = 16
-      const maxCoresPerSocket = pool.cpus.cores
 
       const options = []
-      if (maxCoresPerSocket !== undefined) {
+      if (maxCoresPerSocket !== undefined && vCPUs !== '') {
         for (let coresPerSocket = 1; coresPerSocket <= maxCoresPerSocket; coresPerSocket++) {
-          if (vCPUs !== '' && vCPUs % coresPerSocket === 0 && vCPUs / coresPerSocket <= maxVCPUs) options.push(coresPerSocket)
+          if (vCPUs % coresPerSocket === 0 && vCPUs / coresPerSocket <= maxVCPUs) options.push(coresPerSocket)
```

Comment:
I think the second condition can be eliminated by correctly choosing the initial value of `coresPerSocket` for the loop.